bondcpp/CMakeLists.txt

```diff
@@ -8,6 +8,7 @@ endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
+add_definitions(-DNOMINMAX)
```

Comment:
Do these definitions need to be added per-target rather than project wide?  That way they can be propagated to downstream.

Reply:
I've got no clue ;). @traversaro - do you have any input?

Reply:
So, `-DNOMINMAX` in this case is used is required to make sure that `std::min` contained in the `bond.cpp` (https://github.com/ros/bond_core/blob/5a79fecb3e8b6f2c210d83669b5cb40d47d62c47/bondcpp/src/bond.cpp#L214) works fine, so fortunatly it is nothing in the public headers that needs to be exported. I guess this was added here to be coherent with the earlier use of of `add_compile_options` instead of `target_compile_options` , however I changed it to be per-target in https://github.com/ros/bond_core/pull/83/commits/141dbe1a9cade1112dd62f1a1153cd97e02a04ce .